app.js

```diff
@@ -0,0 +1,342 @@
+// express server
+const express = require('express')
+const app = express()
+app.use(express.json())
+const mongoose = require('mongoose')
+mongoose.set('useFindAndModify', false)
+const { createModel } = require('mongoose-gridfs')
+const { createReadStream, unlinkSync } = require('fs')
+const axios = require('axios')
+
+// general settings
+const config = require('./config.json')
+const boxName = config.nodeName
+const port = config.backendPort
+const dtnd = `${config.dtndIp}:${config.dtndPort}`
+let dtndUuid
+
+const fileUpload = require('express-fileupload')
+app.use(fileUpload({
+  useTempFiles: true,
+  tempFileDir: './tmp/'
+}))
+
+// schemas
+const schemas = require('./schemas')
+
+const contentType = {
+  json: 'application/json'
+}
+
+// check if the request body is in JSON format
+function checkForJSON (req, res, next, expectedType) {
+  if (!req.is(expectedType)) {
+    res.status(415)
+    res.send({ error: 'wrong content type' })
+  }
+  next()
+}
+
+// registers at dtnd rest server and gets the uuid
+async function registerDtnd () {
+  axios({
+    method: 'post',
+    url: `http://${dtnd}/rest/register`,
+    data: {
+      endpoint_id: 'dtn://0/backend'
+    }
+  }).then((response) => {
+    console.log(response.data)
+    if (!response.data.err) {
+      dtndUuid = response.data.uuid
+      console.log('connected to dtnd')
+    } else {
+      console.log('cannot connect to dtnd')
+    }
+  }).catch((err) => {
+    console.log('cannot connect to dtnd', err)
+  })
+}
+
+// Posts Data from Pi to Backend
+// Example:
+// Query: /api/postData/1?format=chunk
+// Body: contains gridFS chunk JSON Data
+// has to be tested
+app.post('/api/postData/:raspberryPiId', (req, res, next) => {
+  checkForJSON(req, res, next, contentType.json)
+},
+async (req, res) => {
+  // connect to raspberryPiId
+  let successfullySent = false
+  const localDB = mongoose.connection.useDb(req.params.raspberryPiId)
+  if (req.query.format === 'chunk') {
+    const chunkModel = localDB.model('fs.chunk', schemas.chunk)
+    await chunkModel.findOneAndUpdate({ _id: req.body._id }, req.body, { upsert: true }).then(() => {
```

Comment:
Why findOneAndUpdate? I thought the chunks and files will be added to this db only one time.

Reply:
Please look up the definition of upsert